src/components/button/types.ts

```diff
@@ -0,0 +1,8 @@
+import { Props } from "../../core/types";
+
+export interface ButtonProps extends Props {
+  type?: string;
```

Comment:
Для таких данных, как тип чего-то, ```string``` не очень подходит. Тип - это ограниченное количество значений. ```string``` допускает любые значения. Тут лучше подойдет использование union'a
```suggestion
  type?: "button" | "submit";
```